opentelemetry-userevents-exporter/src/logs/exporter.rs

```diff
@@ -0,0 +1,252 @@
+            | Severity::Trace3
+            | Severity::Trace4 => eventheader::Level::Verbose,
+
+            Severity::Info
+            | Severity::Info2
+            | Severity::Info3
+            | Severity::Info4 => eventheader::Level::Informational,
+
+            Severity::Error
+            | Severity::Error2
+            | Severity::Error3
+            | Severity::Error4 => eventheader::Level::Error,
+
+            Severity::Fatal
+            | Severity::Fatal2
+            | Severity::Fatal3
+            | Severity::Fatal4 => eventheader::Level::CriticalError,
+
+            Severity::Warn
+            | Severity::Warn2
+            | Severity::Warn3
+            | Severity::Warn4 => eventheader::Level::Warning
+        };
+        let log_es = if let Some(es) = self.provider.find_set(level.as_int().into(), self.exporter_config.get_log_event_keyword())
+        {
+            es
+        } else {
+            return Ok(());
+        };
+        if log_es.enabled() {
+            EBW.with(|eb| {
+                let [mut cs_a_count, mut cs_b_count, mut cs_c_count] = [0 ; 3];
+                let mut eb = eb.borrow_mut();
+                let event_tags: u32 = 0; // TBD name and event_tag values
+                eb.reset(log_data.instrumentation.name.as_ref(), event_tags as u16);
+                eb.opcode(Opcode::Info);
+
+                eb.add_value("__csver__", 0x0401u16, FieldFormat::HexInt, 0);
+                if log_data.record.timestamp.is_some() {
+                    cs_a_count = cs_a_count + 1;
+                }
+                eb.add_struct("PartA", cs_a_count , 0);
+                {
+                    if log_data.record.timestamp.is_some() {
+                        let time: String = chrono::DateTime::to_rfc3339(
+                         &chrono::DateTime::<chrono::Utc>::from(log_data.record.timestamp.unwrap()));
+
+                        eb.add_str("time", time, FieldFormat::Default, 0);
+                    }
+                }
+                // Get Event_Id and Event_Name if present.
+                let (mut event_id, mut event_name) = (0, "");
+                let mut event_count = 0;
+                for (k, v ) in log_data.record.attributes.as_ref().unwrap().into_iter() {
+                    if k.as_str()== "event_id" {
+                        event_id = match v {
+                            AnyValue::Int(value) => {event_count = event_count + 1; *value}
+                            _ => 0
+                        }
+                    }
+                    if k.as_str() == "event_name" {
+                        event_name = match v {
+                            AnyValue::String(value) => {event_count = event_count + 1; value.as_ref()}
+                            _ => ""
+                        }
+                    }
+                }
+                cs_b_count = cs_b_count + event_count;
+                if log_data.record.body.is_some() {
+                    cs_b_count = cs_b_count + 1;
```

Comment:
nit: these bookkeeping of counts can be done at the moment we actually serialize body, to avoid checking same if condition more than once.

Reply:
We need this count value before doing the serialization -to store it in the PartB struct. Another approach would be to have another bool flag to know if the body really exists. That would need multiple such flags for each of the fields. It's tradeoff between memory and performance :)